### PR TITLE
Issue comment can now be configured

### PR DESCRIPTION
**Pull request by muryoh**
This enables custom comment messages that may have hyperlinks
to the issue on the sonar server. It is handy to access the
in a single click to get a better grasp of the context

### CODE REVIEW

File: src/main/java/fr/techad/sonar/GerritPostJob.java [hunk 1]
```diff
@@ -183,8 +181,8 @@ protected ReviewLineComment issueToComment(Issue issue) {
 		ReviewLineComment result = new ReviewLineComment();
 
 		result.setLine(issue.line());
-		result.setMessage(String.format(ISSUE_FORMAT, issue.isNew(), StringUtils.capitalize(issue.ruleKey().toString()),
```
Comment (muryoh) [inline src/main/java/fr/techad/sonar/GerritPostJob.java]: Technically, we're losing the StringUtils.capitalize() here

I hope it's OK. The problem is that sonar server lookups are case sensitive so capitalizing it makes sonar unable to find it back.

File: src/main/java/fr/techad/sonar/gerrit/ReviewUtils.java [hunk 2]
```diff
@@ -119,36 +129,28 @@ public static int maxLevel(ReviewInput reviewInput) {
...
+        HashMap<String, Object> valueMap = new HashMap<>();
+        valueMap.put(prefixKey(ISSUE_PREFIX, ISSUE_IS_NEW), issue.isNew());
+        valueMap.put(prefixKey(ISSUE_PREFIX, ISSUE_RULE_KEY), issue.ruleKey());
+        valueMap.put(prefixKey(ISSUE_PREFIX, ISSUE_SEVERITY), issue.severity());
+        valueMap.put(prefixKey(ISSUE_PREFIX, ISSUE_MESSAGE), issue.message());
+        return substituteProperties(originalMessage, settings, valueMap);
+    }
+
 	public static String substituteProperties(String originalMessage, Settings settings) {
-	    String substitutedString = originalMessage;
```
Comment (muryoh) [inline src/main/java/fr/techad/sonar/gerrit/ReviewUtils.java]: I noticed the substitution had been re-written and was not sure why as the libs we were using were already providing it

I'll put it back if you want me to, just wanted to "lighten the load" :)

Comment (mildis) [inline src/main/java/fr/techad/sonar/gerrit/ReviewUtils.java]: That’s fine : I didn’t know StringUtils provided this functionality (would have save me some time).

File: src/main/java/fr/techad/sonar/gerrit/ReviewUtils.java [hunk 3]
```diff
@@ -119,36 +129,28 @@ public static int maxLevel(ReviewInput reviewInput) {
...
-	        }
-	        if (LOG.isDebugEnabled()) {
-	            LOG.debug("[GERRIT PLUGIN] New message is {}.", substitutedString);
-	        }
-	    } else {
-	        LOG.debug("[GERRIT PLUGIN] No message subtitution to do.");
-	    }
-	
-	    return substitutedString;
+        return substituteProperties(originalMessage, settings, Collections.<String, Object> emptyMap());
```
Comment (mildis) [inline src/main/java/fr/techad/sonar/gerrit/ReviewUtils.java]: I guess declaration of PROP_[START|END|START_LENGHT] can be removed.

File: src/main/java/fr/techad/sonar/GerritConfiguration.java [hunk 4]
```diff
@@ -43,7 +44,8 @@
 	private String changeId;
 	private String revisionId;
 
-	public GerritConfiguration(Settings settings) {
+	public GerritConfiguration(Settings settings)
+	{
```
Comment (mildis) [inline src/main/java/fr/techad/sonar/GerritConfiguration.java]: Opening bracket on the same line than the declaration. (yeah, we’re missing a coding-style file)

**Comment by mildis** [comment]
That sounds good.
However you forgot the localisation files for the new setting you created.

**Comment by muryoh** [comment]
Okay, I'll fix it soon :) Thanks for the feedback

**Comment by mildis** [comment]
Closes #35

**Pull request merged**